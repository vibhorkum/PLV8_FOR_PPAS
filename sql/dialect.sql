CREATE EXTENSION plcoffee;
DO LANGUAGE plcoffee $$ plv8.elog(INFO, "foo") $$;
CREATE EXTENSION plls;
DO LANGUAGE plls $$ plv8.elog(INFO, "foo") $$;

CREATE FUNCTION v8func(a int) RETURNS int[] AS $$
return [a, a, a];
$$ LANGUAGE plv8;

CREATE FUNCTION coffeefunc(a int) RETURNS int[] AS $$
return plv8.find_function('v8func')(a);
$$ LANGUAGE plcoffee;

SELECT coffeefunc(10);

CREATE FUNCTION lsfunc(a int) RETURNS int[] AS $$
return plv8.find_function('v8func')(a);
$$ LANGUAGE plls;

SELECT lsfunc(11);
