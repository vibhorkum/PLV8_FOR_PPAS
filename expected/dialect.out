CREATE EXTENSION plcoffee;
DO LANGUAGE plcoffee $$ plv8.elog(INFO, "foo") $$;
INFO:  foo
CREATE EXTENSION plls;
DO LANGUAGE plls $$ plv8.elog(INFO, "foo") $$;
INFO:  foo
CREATE FUNCTION v8func(a int) RETURNS int[] AS $$
return [a, a, a];
$$ LANGUAGE plv8;
CREATE FUNCTION coffeefunc(a int) RETURNS int[] AS $$
return plv8.find_function('v8func')(a);
$$ LANGUAGE plcoffee;
SELECT coffeefunc(10);
 coffeefunc 
------------
 {10,10,10}
(1 row)

CREATE FUNCTION lsfunc(a int) RETURNS int[] AS $$
return plv8.find_function('v8func')(a);
$$ LANGUAGE plls;
SELECT lsfunc(11);
   lsfunc   
------------
 {11,11,11}
(1 row)

